xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamNavigator.cpp

```diff
@@ -996,12 +996,25 @@ int CDVDInputStreamNavigator::GetSubTitleStreamCount()
 
 int CDVDInputStreamNavigator::GetActiveAudioStream()
 {
-  int activeStream = -1;
+  if (!m_dvdnav)
+  {
+    return -1;
+  }
 
-  if (m_dvdnav)
+  const int8_t logicalAudioStreamId = m_dll.dvdnav_get_active_audio_stream(m_dvdnav);
+  if (logicalAudioStreamId < 0)
+  {
+    return -1;
+  }
+
+  int activeStream = -1;
+  for (int audioN = 0; audioN < GetAudioStreamCount(); audioN++)
```

Comment:
maybe worth to init stream count before the loop
```c
int audioStreamCount = GetAudioStreamCount();
for (int audioN = 0; audioN < audioStreamCount; ++audioN)
```

Reply:
Done, thanks :+1: 
Will also apply to the linked PR